src/main/java/com/flowingcode/vaadin/addons/gridexporter/BaseStreamResourceWriter.java

```diff
@@ -94,28 +95,46 @@ protected Stream<T> getDataStream(Query newQuery) {
     return stream;
   }
 
-  protected List<Pair<String, Column<T>>> getGridHeaders(Grid<T> grid) {
+  protected List<Pair<List<String>, Column<T>>> getGridHeaders(Grid<T> grid) {
     return exporter.getColumnsOrdered().stream()
-        .map(
-            column ->
-                ImmutablePair.of(
-                    renderCellTextContent(grid, column, GridExporter.COLUMN_HEADER), column))
+        .map(column -> ImmutablePair.of(getHeaderTexts(grid, column), column))
         .collect(Collectors.toList());
   }
+  
```

Comment:
@coderabbitai `BaseStreamResourceWriter` and its extending classes is not public API. However the return type must be refactored since we are going to expose BaseStreamResourceWriter in a near future. Please create an issue about it.